tests/data_context/test_data_context_utils.py

```diff
@@ -286,6 +287,174 @@ def test_password_masker_mask_db_url(monkeypatch, tmp_path):
     assert PasswordMasker.mask_db_url("sqlite://", use_urlparse=True) == "sqlite://"
 
 
+def test_password_masker_sanitize_data_context_config(
+    basic_data_context_config,
+    basic_data_context_config_dict,
+    data_context_config_dict_with_cloud_backed_stores,
+    data_context_config_dict_with_datasources,
+    conn_string_password,
+    ge_cloud_access_token,
+):
+    """
+    This unit test ensures that PasswordMasker.sanitize_data_context_config
+    correctly removes passwords from DataContextConfig dicts.
+    """
+
+    # expect that an Exception is raised if something other than a dict is passed
+    with pytest.raises(TypeError):
+        PasswordMasker.sanitize_data_context_config(basic_data_context_config)
+
+    # expect no change without datasources
+    config_without_creds = PasswordMasker.sanitize_data_context_config(
+        basic_data_context_config_dict
+    )
+    assert config_without_creds == basic_data_context_config_dict
+
+    # test that cloud store backend tokens have been properly masked
+    config_with_creds_in_stores = PasswordMasker.sanitize_data_context_config(
+        data_context_config_dict_with_cloud_backed_stores
+    )
+    for name, store_config in config_with_creds_in_stores["stores"].items():
+        try:
+            # check that the original token exists
+            assert (
+                data_context_config_dict_with_cloud_backed_stores["stores"][name][
+                    "store_backend"
+                ]["ge_cloud_credentials"]["access_token"]
+                == ge_cloud_access_token
+            )
+            # expect that the GE Cloud token has been obscured
+            assert (
+                store_config["store_backend"]["ge_cloud_credentials"]["access_token"]
+                != ge_cloud_access_token
+            )
+        except KeyError:
+            # expect config to not be changed
+            assert (
+                store_config
+                == data_context_config_dict_with_cloud_backed_stores["stores"][name]
+            )
```

Comment:
Although this is definitely a unit test, I feel like it's pretty complicated and is testing a lot of different things. Could we simplify this at all? Maybe a snapshot test or splitting it up into a few different tests?

Having `try/except` and `if/else` for various asserts makes debugging this a little difficult imo.

Reply:
thanks for the feedback -- i'll break these cases out into separate tests.

Reply:
I've factored the tests out into their own functions, but have left the try/except, since it's the most straightforward way to iterate through unknown values and check their configuration. Let me know if this is easier to parse!